### PR TITLE
Replace ISOLATED_INSTANCE with APP_ENV

### LINKED ISSUE #824: Hydration errors in static pages when using the ISOLATED_INSTANCE env due to react tree and dom being out of sync
Currently in staging there are hydration errors when you navigate to the home page.

image

Hydration errors are appearing for static pages when running the application in production mode (yarn build && yarn start) or having the `ISOLATED_INSTANCE` set when running in development mode (yarn dev). This is because during the build process the static pages are built without content due to the ISOLATED_INSTANCE env var. The ISOLATED_INSTANCE environment var was needed in order to prevent server content being generated when no database was available.

Logic needs to be reworded to get rid of the hydration errors and ensure that static pages where content is available are properly built

**Acceptance Criteria**:
- [ ] Static pages load without generating hydration errors when the `ISOLATED_INSTANCE` is set

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

Closes #824 

Reworks the logic to remove dependence on the ISOLATED_INSTANCE flag when building the Next JS application.  Static pages were being generated with the ISOLATED_INSTANCE flag during the build process which blocked the passing of child components to the Server Side Rendering.

A new APP_ENV is introduced that clearly indicates the application is in test mode.  The APP_ENV=test for Cypress and Jest testing.  This variable ensures that initial flags are used even if a REDIS_URL is present in the environment.

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [ ] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/api/submit.ts [hunk 1]
```diff
@@ -278,6 +278,14 @@ const processFormData = async (
       }`
     );
 
+    // Do not process if in TEST mode
+    if (process.env.APP_ENV === "test") {
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply pages/api/submit.ts:282]: You're right, when I put in the short circuit I wasn't thinking about Jest coverage (which already mocks pushFileToS3 and callLambda), only Cypress (which you can't mock external modules).  Will find another way.

**Review by Moro-Code — COMMENTED** [review]
Wondering if it would be better to remove a variable and just use a value of `test` for NODE_ENV

File: .env.example [hunk 2]
```diff
@@ -39,8 +39,7 @@ TEMPORAY_TOKEN_TEMPLATE_ID=b6kdhaud-d10a-422a-973f-05e274d9aa86
...
 TEMPLATE_ID=8pa6247s75-a1d6-4e3c-8421-042a2b4158b7
 
-# Used when no Redis instance is available
+# Used when no Redis instance or Database is available
 # The application returns all default flags values when this variable is set.
 # This variable is automatically set when running tests (Jest + Cypress)
-# and during the NextJS build process when prerendering pages
-ISOLATED_INSTANCE=true
\ No newline at end of file
+# APP_ENV=test
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply .env.example:45]: Unfortunately in nextJS you can not override NODE_ENV when running next dev or next start.  The only valid values it accepts are "development" and "production".  That was my first go to move as well.
https://github.com/vercel/next.js/issues/17032

File: pages/api/submit.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply pages/api/submit.ts:282]: Currently in all tests the `submitToReliability` queue flag is off so it always bypasses that block of code.  I just added an early return earlier in the function to bypass the call to the Template Lambda which returned the wrong formConfig for what was being tested in Cypress.

**Review by Moro-Code — APPROVED** [review]

**Pull request merged**